### PR TITLE
Multitable and Tabbook fix

### PR DESCRIPTION
**Pull request by jonkeane**
Adds checks for multitables and tabbooks:
* multitables can no longer be set when using the deprecated `selected_array` method. The error that is thrown explains what's going on and how to unset that option.
* Tabbooks can no longer be made with a multitable that uses the deprecated `selected_array` method. After the upcoming migration of `selected_array` to `as_selected`, none of these should exist, however just in case: the error explains what's going on and how to unset that option.
* both `selected_array` and `as_selected` methods for MR will be sent in the proper query format of `[each: ..., function: ...]`

### CODE REVIEW

File: R/tab-book.R [hunk 1]
```diff
@@ -158,18 +168,7 @@ setMethod("initialize", "MultitableResult", function (.Object, ...) {
...
-        ## that column is multitable var (3 -> 2), row is category of
-        ## array (2 -> 1), subvar is "tab" (1 -> 3)
-        if (length(dim(cube)) == 3L) {
-            ## TODO: CrunchCube should get a dimensions<-
-            cube@dims <- CubeDims(cube@dims[c(2, 3, 1)])
-            cube@arrays <- lapply(cube@arrays, aperm, perm=c(2, 3, 1))
-        }
-        return(cube)
-    })
+    .Object$result <- lapply(.Object$result, CrunchCube)
```
Comment (nealrichardson) [inline R/tab-book.R]: Why is this deleted? This is about categorical array, not multiple response.

**Comment by jonkeane** [comment]
This is ready for review again. I've tested it against `crunctabs`'s subtotals branch (which @persephonet pointed me towards) with the following and everything worked without error producing reasonable looking files, modulo a few known issues (eg varying bases with MRs).

The display methods for tabbooks will need some work, but I think we can defer that for now.

```
setupCrunch("app")
library(crunch)
library(crunchtabs)
login()

ds <- loadDataset("Example dataset")

# subset dataset to exclude types incompatible with crunchtabs
sub_ds <- ds[!(types(variables(ds)) %in% c("text", "datetime"))]

# toplines work well
tst <- crosstabs(sub_ds)
writeLatex(tst, filename = 'tst')
writeExcel(tst, filename = 'tst')

#categorical banner
q1_banner <- banner(sub_ds, vars = list(`banner 1` = c('q1')))
# MR banner
allpets_banner <- banner(sub_ds, vars = list(`banner 1` = c('allpets', 'q1')))

# crosstabs with a single categorical banner
crosstabs_summary <- crosstabs(dataset = sub_ds, banner = q1_banner)
writeLatex(crosstabs_summary, filename = 'tst_q1')
writeExcel(crosstabs_summary, filename = 'tst_q1')

# crosstabs with an MR and a categorical banner
crosstabs_summary <- crosstabs(dataset = sub_ds, banner = allpets_banner)
writeLatex(crosstabs_summary, filename = 'tst_allpets')
writeExcel(crosstabs_summary, filename = 'tst_allpets')
```

File: R/tab-book.R [hunk 2]
```diff
@@ -158,18 +168,36 @@ setMethod("initialize", "MultitableResult", function (.Object, ...) {
                 name="Total"
             )
         )))
+    # .Object$result <- lapply(.Object$result, CrunchCube)
```
Comment (nealrichardson) [inline R/tab-book.R]: Remove

File: R/tab-book.R [hunk 3]
```diff
@@ -158,18 +168,36 @@ setMethod("initialize", "MultitableResult", function (.Object, ...) {
...
+            ## underlyingly, not 3 dims
+            selecteds <- is.selectedDimension(cube@dims)
+            if (any(which(selecteds) %in% c(3, 4))) {
+                ## the selected dimension is in the second half of the cube, so 
+                ## the MR is in the multitable
+                cube@dims <- CubeDims(cube@dims[c(2, 3, 4, 1)])
+                cube@arrays <- lapply(cube@arrays, aperm, perm=c(2, 3, 4, 1))
+            } else if (any(which(selecteds) %in% c(1, 2))) {
+                ## the selected dimension is in the first half of the cube, so 
+                ## the array is in the multitable
```
Comment (nealrichardson) [inline R/tab-book.R]: Is it possible for a categorical array to be in the multitable? I don't know how I'd interpret that.

File: inst/app.crunch.io/api/datasets/1/multitables/apidocs-ca-mr-tabbook.json [hunk 4]
```diff
@@ -0,0 +1 @@
...e},{"numeric_value":9,"id":9,"name":"Not Asked","missing":true}]}},"missing":false},{"id":2,"value":{"derived":false,"references":{"alias":"petloc_work","name":"Work"},"id":"0002","type":{"ordinal":false,"class":"categorical","categories":[{"numeric_value":1,"id":1,"name":"Cat","missing":false},{"numeric_value":2,"id":2,"name":"Dog","missing":false},{"numeric_value":3,"id":3,"name":"Bird","missing":false},{"numeric_value":8,"id":8,"name":"Skipped","missing":true},{"numeric_value":9,"id":9,"name":"Not Asked","missing":true}]}},"missing":false}],"class":"enum"}},{"derived":false,"references":{"alias":"petloc","description":"Name the kinds of pets you have at these locations.","name":"Pets by location","subreferences":[{"alias":"petloc_home","name":"Home"},{"alias":"petloc_work","name":"Work"}]},"type":{"ordinal":false,"subvariables":["0001","0002"],"class":"categorical","categories":[{"numeric_value":1,"id":1,"name":"Cat","missing":false},{"numeric_value":2,"id":2,"name":"Dog","missing":false},{"numeric_value":3,"id":3,"name":"Bird","missing":false},{"numeric_value":8,"id":8,"name":"Skipped","missing":true},{"numeric_value":9,"id":9,"name":"Not Asked","missing":true}]}}],"n":20}}]}]}
```
Comment (nealrichardson) [inline inst/app.crunch.io/api/datasets/1/multitables/apidocs-ca-mr-tabbook.json]: Would you mind prettifying this JSON? (Been meaning to look into how to make `httptest` write out pretty JSON without parsing in R, but it's worth doing manually here.)

File: R/tab-book.R (hunk 3 above)
Comment (jonkeane) [reply R/tab-book.R]: Yup 🙃!
